src/backend/pipeline/cont_execute.c

```diff
@@ -319,6 +338,63 @@ InsertBatchAckTuple(InsertBatchAck *ack)
 		pg_atomic_fetch_add_u32(&ack->batch->num_cacks, 1);
 }
 
```

Comment:
I found InsertBatchAckCreate a little hard to read. Maybe put in a one-liner comment.